website/blog/2024-01-17-running-tabby-as-a-language-server/index.md

```diff
@@ -0,0 +1,61 @@
+---
+slug: running-tabby-as-a-language-server
+title: "Running Tabby as a Language Server"
+authors: [icycodes]
+tags: []
+---
+
+I am excited to announce that a new feature has been added to Tabby, allowing you to run it as a [language server](https://microsoft.github.io/language-server-protocol/).
+
+Previously, Tabby provided extensions for popular editors like [VSCode](https://marketplace.visualstudio.com/items?itemName=TabbyML.vscode-tabby), [IntelliJ Platform IDEs](https://plugins.jetbrains.com/plugin/22379-tabby), and [VIM/NeoVIM](https://github.com/TabbyML/vim-tabby), which offered inline code completion. These plugins were all based on the same client agent called [tabby-agent](https://github.com/tabbyml/tabby/tree/main/clients/tabby-agent/), which implemented features such as caching, debouncing, and post-processing to enhance code completion. The tabby-agent communicated with the IDE via a customized protocol based on JSON Lines, originally designed for compatibility with VIM's JSON mode channel. By adding support for the Language Server Protocol, Tabby should be able to provide a more flexible and welcoming experience to contributors.
+
+## Running Tabby as a Language Server
+
+To run Tabby as a language server, you'll first need to follow [this guide](https://tabby.tabbyml.com/docs/installation/) to set up your Tabby server. Once your Tabby server is up and running, you can proceed with installing the `tabby-agent` package.
+
+The `tabby-agent` has been released on [npm](https://www.npmjs.com/package/tabby-agent) as a separate package, making it easy to install and run as a language server. To get started, ensure that you have [Node.js](https://nodejs.org/) v18 or above installed, and then execute the following command:
+
+```bash
+npx tabby-agent --lsp --stdio
```

Comment:
Please include an example screenshot showing what it looks like on command line. There is currently no output when the command is run, and I'm unsure if that means it's working or not. Could some output be added to the process so the user knows whether it's running or not? (Separate PR)

Reply:
As the language server listens on StdIO, it's better not to output messages to the user. 
To provide more clarity, I have added additional explanations and a GIF to guide users on manually killing the server after test running.